kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -189,40 +240,30 @@ def after_catalog_created(self, catalog):
         )
 
 
-def _is_known_ci_env(known_ci_env_var_keys=KNOWN_CI_ENV_VAR_KEYS):
+def _is_known_ci_env(known_ci_env_var_keys: set[str]):
     # Most CI tools will set the CI environment variable to true
     if os.getenv("CI") == "true":
         return True
     # Not all CI tools follow this convention, we can check through those that don't
     return any(os.getenv(key) for key in known_ci_env_var_keys)
 
 
-def _get_project_properties(user_uuid: str, project_path: Path) -> dict:
-    hashed_package_name = _hash(str(PACKAGE_NAME)) if PACKAGE_NAME else "undefined"
+def _get_project_properties(user_uuid: str, pyproject_path: Path) -> dict:
+    project_uuid = _get_or_create_project_uuid(pyproject_path)
+    package_name = PACKAGE_NAME or UNDEFINED_PACKAGE_NAME
```

Comment:
Does this mean that all packaged projects will have `project_uuid = "undefined_project_uuid"`? Isn't it better to send a `None` to Heap so we can outright ignore those?

Reply:
We send `_hash(f"{project_uuid}{package_name}")` to the Heap. So we are still able to differentiate those projects by `package_name`. But if it's not the desired behaviour and we want to explicitly filter out projects for which UUID were not defined - then I can add that, so we send None instead.

Reply:
Yeah, since we're salting `package_name` with `project_uuid`, in case the `project_uuid` is not found we should send `None` directly, to avoid accidentally using the same salt with 2 different package names

Reply:
Done